### PR TITLE
Test for RSTensor send and get. Also added equal operation for RSTensor

### PR DESCRIPTION
**Pull request by kamathhrishi**
## Description
Added test for sending and getting RSTtensor which will pass after merging PR for RSTensor proto definition https://github.com/OpenMined/PySyft/pull/5598. 

## How has this been tested?
Test to check send and get functionality for RSTensor 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/sympc/tensor/replicatedshare_tensor.py [hunk 1]
```diff
@@ -129,6 +132,14 @@ def eq(self, y):
 
         """
 
+        if not (torch.cat(self.shares) == torch.cat(y.shares)).all():
```
Comment (gmuraru) [inline src/sympc/tensor/replicatedshare_tensor.py:135]: Q: Why it is needed to call `torch.cat`? and not simply do `if not (self.shares == y.shares)`

Comment (kamathhrishi) [reply src/sympc/tensor/replicatedshare_tensor.py:135]: It was throwing an error `E       RuntimeError: Boolean value of Tensor with more than one value is ambiguous` so I cast each of them into a torch tensor and performed the operation. self.shares and y.shares are a list of tensors, now they become 2d torch tensors after using cat.

**Review by rasswanth-s — APPROVED** [review]
Looks good to me.

**Review by rasswanth-s — APPROVED** [review]
Looks good to me.

**Pull request merged**